build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBaseClient.java

```diff
@@ -86,4 +101,39 @@ public void setLog(Log log) {
     public String getArtifactoryUrl() {
         return artifactoryUrl;
     }
+
+    public ArtifactoryVersion getArtifactoryVersion() {
+        if (artifactoryVersion == null) {
+            try {
+                artifactoryVersion = httpClient.getVersion();
+            } catch (IOException e) {
+                artifactoryVersion = ArtifactoryVersion.NOT_FOUND;
+            }
+        }
+        return artifactoryVersion;
+    }
+
+    public boolean isRepoExist(String repo) {
+        String fullItemUrl = artifactoryUrl + API_REPOSITORIES + "/" + repo;
+        String encodedUrl = ArtifactoryHttpClient.encodeUrl(fullItemUrl);
+        HttpRequestBase httpRequest = new HttpGet(encodedUrl);
+        HttpResponse httpResponse = null;
+        int connectionRetries = httpClient.getConnectionRetries();
+        try {
+            httpResponse = httpClient.getHttpClient().execute(httpRequest);
+            StatusLine statusLine = httpResponse.getStatusLine();
+            if (statusLine.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
+                return false;
+            }
+        } catch (IOException e) {
+            return false;
```

Comment:
I think it is better to remove this catch and allow this method to throw the exception.